### DISCUSSION TITLE
Set digital output time

### DISCUSSION THREAD

**Discussion by Sigmalalala** [bonsai-rx/bonsai#1185] [Q&A]
Hello! I have two workflows here. The first one is that when the subject moves to the specific area, the arduino will at 5V output; the second is that the arduino will be 5V output for 2 seconds. But I don't know how to combine these two workflows. What I want to achieve is that when the subject moves to the position I am interested in, I will give it a 2-second electric shock through the arduino connected with the shocker. I really need help and I hope you can help me. I will be very grateful!
![微信图片_20221211214012](https://user-images.githubusercontent.com/118099060/206907000-0f129c30-b3d5-4395-aeba-58d05570c24d.png)
![微信图片_20221211214018](https://user-images.githubusercontent.com/118099060/206907006-0a472e43-abee-41fa-9920-b2dcf52fd923.png)

**Comment by kapilsaxena33** [comment]
There is something like this shown there, give it a try https://bonsai-rx.org/docs/tutorials/state-machines.html

**Comment by kapilsaxena33** [comment]
I think your digital output (in the first diagram) should be replaced by the workflow that contains the on and off stimulus.

**Reply by Sigmalalala** [reply]
I have tried, but it did not work. Thank you for your answer! So nice of you.
